### PR TITLE
Add attribute hooks to job related components 

### PR DESCRIPTION
**Pull request by marlovr**
**Release Type:** *Dev Improvement* 

Fixes https://x-team-internal.atlassian.net/browse/XP-3689

## Description

This PR was made to clean up https://github.com/x-team/xp-ui/pull/539 which had a missing `data-testid` attribute which was causing the 'Successfully Withdraw a job application' test in https://github.com/x-team/xp-registration/pull/192 to fail. 

## Checklist

**Before submitting a Pull Request,** please make sure the following is done:

- [x] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `PEER REVIEW` column
- [x] Flow typechecks passed (`npm run flow`)
- [x] Snapshots tests passed (`npm run jest`) (**NOTE:** This is temporarily disabled. Skip it.)
- [x] check cleanup tasks (https://github.com/x-team/xp/labels/cleanup) and take a suitable small one (if exists) in a related area of the current changes
- [ ] if added a new UI component that is meant to be used directly by other apps, remember to export that from the appropriate bundle file: `src/index.js` and/or `src/registration.js`
- [ ] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases and/or debug cases
- [ ] if you've fixed a bug, make sure to also include a new story that will expose declarations with problematic data that caused the bug in the first place, so that we can assure that no regressions will pop up on future interactions
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot (**NOTE:** This is temporarily disabled. Skip it.)
- [x] **manually tested the app** by running it in several different browsers (Firefox, Chrome, Opera, Safari, MS IE/Edge, etc.) and checked nothing is broken and operates as expected!

## Related PRs

branch | PR
------ | ------
XP-3689-Add-data-testid-attribute-to-job-related-components-1 | [XP UI](https://github.com/x-team/xp-ui/pull/539)
Xp-3623-Cypress-E2E-View-Jobs-Board | [XP Registration](https://github.com/x-team/xp-registration/pull/192)

## Steps to Test or Reproduce

n/a
## Impacted Areas in Application

`src/components/ui/Modal.js`

### CODE REVIEW

File: src/components/ui/HeaderBar.js [hunk 1]
```diff
@@ -286,7 +286,7 @@ class HeaderBar extends PureComponent<Props, State> {
             <div className={cx.logo}>
               {xTeamBrand(78, 32, theme.typoHeading)}
             </div>
-            <nav className={[cx.nav, expanded ? cx.expandedNav : ''].join(' ')}>
+            <nav className={[cx.nav, expanded ? cx.expandedNav : ''].join(' ')} data-testid='xpui-headerbar-nav'>
```
Comment (kararade) [inline src/components/ui/HeaderBar.js:289]: hey @marlovr 

should `xpui-headerbar-nav` be `xpui-headerBar-nav` using camelCase?

Correct if I'm wrong 😉 

Thanks

File: src/components/ui/WelcomeHero.js [hunk 2]
```diff
@@ -186,9 +186,9 @@ type Props = {
 }
 
 const WelcomeHero = ({ heading, videoUrl, onDismiss, children }: Props) => heading || children ? (
-  <div className={cx.wrapper}>
+  <div className={cx.wrapper} data-testid='xpui-welcome-hero-wrapper'>
     <div className={cx.container}>
-      <div className={cx.content}>
+      <div className={cx.content} data-testid='xpui-welcome-hero-content'>
```
Comment (kararade) [inline src/components/ui/WelcomeHero.js:191]: hey @marlovr 

Here you are using `xpui-welcome-hero-*` where the filename is `WelcomeHero.js`, do you think we can keep the consistency in the naming changing these `xpui-welcome-hero-*` to `xpui-welcomeHero-*` as the `xpui-jobDetails-heading` in the `JobDetails.js` or `xpui-jobCard` in the `JobCard.js` files? 

Thanks

File: src/components/ui/HeaderBar.js (hunk 1 above)
Comment (marlovr) [reply src/components/ui/HeaderBar.js:289]: You are absolutely correct! Thank you for picking up on that. :)

File: src/components/ui/WelcomeHero.js (hunk 2 above)
Comment (marlovr) [reply src/components/ui/WelcomeHero.js:191]: Yes, thanks again for picking up on that. I'll fix that up.

**Comment by marlovr** [comment]
I've addressed the changes requested and updated the status. :)

**Comment by nicksp** [comment]
There were several changes made recently to the `develop` branch of the `xp-ui` so I went ahead of making a rebased and appropriate changes. Looks good to me now and going to move this forward. Great work @marlovr

**Pull request merged**